source/PlayerInfo.cpp

```diff
@@ -385,7 +385,11 @@ void PlayerInfo::AddChanges(list<DataNode> &changes)
 		// Recalculate what systems have been seen.
 		GameData::UpdateSystems();
 		seen.clear();
-		double jumpRange = (flagship && flagship->JumpRange()) ? flagship->JumpRange() : System::DEFAULT_NEIGHBOR_DISTANCE;
+		double jumpRange;
+		if(flagship)
+			jumpRange = flagship->JumpRange();
```

Comment:
What if the JumpRange is lower than System::DEFAULT_NEIGHBOR_DISTANCE? Don't we want the planets within default_neighbor_distance to be also seen in that case? (Same as when the player had no jumpdrive.)

Reply:
The way I'm looking at it is that if the player can jump to it, the player can see it. If the player can't jump to it, then the player can't see it.

Reply:
It feels strange to me that a player without a jumpdrive can see systems further away than a player with a "smaller-than-normal-range" jumpdrive.
Somehow I would expect a minimum vision distance to always apply (even if you do have a jump drive and cannot jump to it). And if you then want to hide systems, then hide them at a "just-beyond-normal-range" location.

Reply:
What if the system itself has a jump range smaller than the default? Should that also be ignored for the view range?

Reply:
Right now the ring drawn on the map is the jump range, but the comments in that section of the code describe it as the visible systems range. Should a second ring be drawn for the visible range if the jump and visible range differ (i.e. if the player's jump range is less than 100 due to their jump drive)?

Reply:
> What if the system itself has a jump range smaller than the default? Should that also be ignored for the view range? 

Yes, I feel this should also use the default view-range.

> Should a second ring be drawn for the visible range if the jump and visible range differ (i.e. if the player's jump range is less than 100 due to their jump drive)?

Sounds appropriate, for the system that the player is in. But I can imagine that having 2 rings can be confusing, so it would also be okay for me if in that case only the jump-range ring is drawn for the system that the player is in. (Because the player will be more interested in where he/she can jump to. The player will see the systems anyway.)

Reply:
> Yes, I feel this should also use the default view-range.

I’m not so sure on that, as the argument for having the view range be 100 if the player’s jump drive has a lesser range than that doesn’t also apply to if the system’s jump range is lesser. That is, in the situation where the player’s JD range is less than 100, they could remove the JD to increase their view range (which admittedly doesn’t make much sense), but in the case of the system’s jump range being lesser, whether the player does or does not have a JD doesn’t affect the view range, as it is always static and set by the system.

Reply:
Ok, when the system has a jump-range smaller than the view-range, then I would also be okay with the visual range being reduced to the smaller jump-range.